src/cargo/ops/cargo_test.rs

```diff
@@ -80,10 +81,15 @@ fn run_unit_tests(
 
     let mut errors = Vec::new();
 
-    for &(ref pkg, ref kind, ref test, ref exe) in &compilation.tests {
+    for &(ref pkg, ref target, ref exe) in &compilation.tests {
+        let kind = target.kind();
+        let test = target.name().to_string();
         let exe_display = exe.strip_prefix(cwd).unwrap_or(exe).display();
         let mut cmd = compilation.target_process(exe, pkg)?;
         cmd.args(test_args);
+        if target.harness() && options.quiet_run {
```

Comment:
Sorry, just one more minor question.  Is there a reason `quiet_run` was added to `TestOptions` instead of just checking the verbosity level?  Something like: `config.shell().verbosity() == Verbosity::Quiet`.  The reason I ask is because "cargo -q test" behaves differently from "cargo test -q".

Reply:
Oops! Had missed this. I've updated changes. Thanks again.